### DISCUSSION TITLE
Only adding text if it'll fit

### DISCUSSION THREAD

**Discussion by jranieri-grammatech** [moshi4/pyCirclize#25] [Q&A] [answered]
<img width="632" alt="j8" src="https://github.com/moshi4/pyCirclize/assets/47784968/f5bb8a0d-37d4-478f-ada5-9ed09062d9f9">

I've been using pyCirclize to graph performance of an internal tool. The light blue rectangles represent files the tool is working on and the text 'inside' is the file name. As seen in the image, the file names tend to overlap and cause a mess.

I'd like the text only to be added if it'll fit in the rectangle. Is there a good way to do that?

**Comment by moshi4** [comment, marked as answer]
Hi @jranieri-grammatech,

pyCirclize does not provide a method to display text only if it fits within a rectangle.
I also think that implementing such a method would probably be very difficult.
I am sorry, but I am not likely to be able to provide a solution at this time.

**Reply by jranieri-grammatech** [reply]
Thanks for taking a look at this and confirming that it's tricky problem. For now I'll use some simple heuristics to try to clean this up.
